vendor/spawn/src/spawn_stubs.c

```diff
@@ -29,7 +29,7 @@ CAMLprim value spawn_is_osx()
 
 #include <assert.h>
 #include <string.h>
-#if !defined(__CYGWIN__)
+#if !defined(__CYGWIN__) && !defined(__HAIKU__)
```

Comment:
Could you make this change in this repo https://github.com/janestreet/spawn

Once it's accepted there, it will automatically be applied here.

Reply:
Thanks. Could you remove the change from this PR then? We can merge the command line flags.

Reply:
This change was also reverted upstream, so it might have to be submitted again.